### PR TITLE
libCEED: Update for v0.9.0

### PR DESCRIPTION
**Pull request by jedbrown**

### CODE REVIEW

**Comment by jedbrown** [comment]
Huh, this relocation error looks peculiar to me @pazner.
```
[03:32:15] /opt/x86_64-unknown-freebsd11.1/bin/x86_64-unknown-freebsd11.1-ld: build/interface/ceed-elemrestriction.o: unrecognized relocation (0x2a) in section `.text'
[03:32:15] /opt/x86_64-unknown-freebsd11.1/bin/x86_64-unknown-freebsd11.1-ld: final link failed: Bad value
```
https://dev.azure.com/JuliaPackaging/Yggdrasil/_build/results?buildId=12152&view=logs&j=5532483c-82b1-57c6-5c2d-67107c64ceb3&t=761e90bc-85b6-543d-3c4d-0bdf7616b6c1&l=139

**Comment by pazner** [comment]
I noticed that #3278 is failing with the same error. Perhaps it's an issue with the CI?

**Comment by giordano** [comment]
The issue is https://github.com/JuliaPackaging/BinaryBuilderBase.jl/issues/158. As a temporary work around you can pass `preferred_llvm_version=v"11"` to `build_tarballs`

**Comment by jedbrown** [comment]
Tests pass with the workaround. Thanks.

File: L/libCEED/build_tarballs.jl [hunk 1]
```diff
@@ -30,4 +30,4 @@ products = [
 dependencies = Dependency[]
 
 # Build the tarballs, and possibly a `build.jl` as well.
-build_tarballs(ARGS, name, version, sources, script, platforms, products, dependencies)
+build_tarballs(ARGS, name, version, sources, script, platforms, products, dependencies, preferred_llvm_version=v"11")
```
Comment (giordano) [inline L/libCEED/build_tarballs.jl:33]: Nevermind, with #3288 it shouldn't be necessary anymore:
```suggestion
build_tarballs(ARGS, name, version, sources, script, platforms, products, dependencies)
```
Let's try

**Pull request merged**